xpdacq/xpdacq.py

```diff
@@ -127,8 +129,10 @@ def validate_dark(light_cnt_time, expire_time, dark_scan_list = None):
         dark_field_uid : str
             uid to qualified dark frame
     '''
-    if not dark_scan_list: dark_scan_list= _read_dark_yaml()
-    if len(dark_scan_list) > 0:
+    if not dark_scan_list:
+        dark_scan_list= _read_dark_yaml()
+    #if len(dark_scan_list) > 0:
```

Comment:
why are you changing this line?  It is testing something different...it is testing that the list exists but is empty, not if it doesn't exist.

I am not sure that I really like your fix of this bug.  The bug was really that something went wrong during the initialization sequence of icollection, so I would rather that the bug-fix fixed the initialization.  In other words, something happens to create the dark yaml file during start beamtime sequence.    I would definitely feel much more comfortable with that.

Having _read_yaml crash when there is no yaml is actually better behavior in a way because it lets us know that something is wrong with the intialization (as it did here) that we need to fix.  Maybe we find other problems with the initialization logic when we do that.